bindings/java/fdbJNI.cpp

```diff
@@ -612,14 +612,14 @@ JNIEXPORT jobject JNICALL Java_com_apple_foundationdb_FutureMappedResults_Future
 		FDBMappedKeyValue kvm = kvms[i];
 		int kvm_count = kvm.getRange.m_size;
 
-		// now it has 5 field, key, value, getRange.begin, getRange.end, boundaryAndExist
+		// now it has 5 field, key, value, getRange.begin, getRange.end
 		// this needs to change if FDBMappedKeyValue definition is changed.
 		const int totalFieldFDBMappedKeyValue = 5;
 
 		const int totalLengths = totalFieldFDBMappedKeyValue + kvm_count * 2;
 
 		int totalBytes = kvm.key.key_length + kvm.value.key_length + kvm.getRange.begin.key.key_length +
-		                 kvm.getRange.end.key.key_length + sizeof(kvm.boundaryAndExist);
+		                 kvm.getRange.end.key.key_length;
```

Comment:
How safe is this change across different code versions? (I'm thinking, for example, a new version of the C bindings being used with an older Java client, which is a supported configuration, or more general uses of the multi-version client.) I can see how it's not all that useful, but does more care need to be taken to make sure that we don't break those cases?

Reply:
good point, is there a best practice for this type of change? maybe remove java first, and then in the next release remove c? also cc @jzhou77

Reply:
Such a change is incompatible across versions. Not sure about how JNI should handle such cases, but in C bindings, we do have versioning for APIs. I guess the best solution is to look at API version and react differently, but I am not sure if that's possible here.